### PR TITLE
[Product CRUD] External Products

### PR DESCRIPTION
**Pull request by emdashcodes**
Here's a first pass at implementing CRUD for the external product type. Includes some tests so we can make sure both the general product data (name, price, etc) and type specific data is updated and read correctly.

### CODE REVIEW

File: includes/class-wc-product-external.php [hunk 1]
```diff
@@ -65,23 +146,37 @@ public function add_to_cart_text() {
...
+	 * Helper method that updates all the post meta for an external product.
 	 *
-	 * @access public
-	 * @return string
+	 * @since 2.7.0
+	 * @param int $id
 	 */
-	public function get_button_text() {
-		return $this->button_text ? $this->button_text : __( 'Buy product', 'woocommerce' );
+	protected function update_post_meta( $id ) {
```
Comment (justinshreve) [inline includes/class-wc-product-external.php]: I've noticed in all the other CRUD classes, we accept an `$id` here. I can't think of any case where we would call update_post_meta and pass a different ID. We can just use `get_id`. Should we go ahead and do a PR to update that for all the other CRUD classes?

**Comment by justinshreve** [comment]
Putting this back at in progress because I'm opening a PR to work on a couple of the abstract things and will then rebase this one.

File: includes/class-wc-product-external.php (hunk 1 above)
Comment (claudiosmweb) [inline includes/class-wc-product-external.php]: Since is protected I don't see why use an `$id` too.

**Review by claudiosanches — APPROVED** [review]
Looks good to me :)

**Pull request closed without merging**